bin/fastq_scan.py

```diff
@@ -258,8 +312,53 @@ def args_in():
     parser.add_argument("-p", "--high-precision", help="Enable high precision arithmetic", default=False, type=str2bool)
     args_out = parser.parse_args()
     return args_out
-
+    
 
 if __name__ == "__main__":
     args = args_in()
-    FastQReader(args.files, args.names, args.high_precision)
+
+
+    # Verify some files were passed in
+    if not len(args.files):
+        sys.stderr.write("No files specified\n")
+        sys.exit(1)
+
+    # Set names if none are provided
+    names = args.names
+    if args.names is None:
+        names = [idx for idx in range(len(args.files))]
+
+    # Set the function used for numeric percision
+    numeric_conversion = float
+    if args.high_precision:
+        numeric_conversion = decimal.Decimal
+
+    fq_data: dict[str, float|decimal.Decimal] = {}
+    combined_data = FastQData("combined", float)
+    for file, name in zip(args.files, names):
+        file_p = p.Path(file)
+        data = FastQData(name, float)
+        with get_file_reader(file_p) as fastq:
+            for line in fastq:
+                try:
+                    header = line.strip()
+                    sequence = next(fastq).strip()
+                    plus = next(fastq).strip()
+                    quality = next(fastq).strip()
+                except StopIteration:
+                    pass
+                else:
+                    verify_fastq(file, header, sequence, plus, quality)
+                    data.add_record(sequence, quality)
+        fq_data[data.name] = data.to_dict()
+        combined_data += data
+        sys.stderr.write(f"Finished reading file: {name}\n")
+
+    fq_data[combined_data.name] = combined_data.to_dict() 
+    print(json.dumps(fq_data, indent=2))
```

Comment:
Do you want it printed to stdout or should this be put into a file?

Reply:
I could take a look at the module. Is it this one? https://github.com/phac-nml/mikrokondo/blob/optimization/fastqscan/modules/local/fastqscan.nf